File: server/services/suscripcion.api/models/classes/boleto.js
/** 
 * Clase del modelo de Boleto cuyos atributos se adjuntarán al modelo de Suscripción para el envío de correos electrónicos de suscripción a Wallets.
 * @class Boleto
 * @constructor
 * @param { any } boleto Datos del boleto
 * @description Modelo de Boleto.
 * - Atributos:
 * @attribute { numeric } viaje_id | Identificador del viaje. Ejemplo: 1
 * @attribute { numeric } boleto_id | Identificador del boleto. Ejemplo: 1
 * @attribute { string } correo_pasajero | Correo del pasajero. Ejemplo: "ejemplo@gmail.com"
 * @attribute { string } nombre_pasajero | Nombre del pasajero. Ejemplo: "Juan Perez"
 * @attribute { integer } numero_asiento | Numero de asiento asignado. Ejemplo: 23
 * @attribute { string } origen | Origen. Ejemplo: "GUADALAJARA"
 * @attribute { string } destino | Destino. Ejemplo: "CDMX"
 * @attribute { integer } numero_servicio | Numero de servicio. Ejemplo: 123
 * @attribute { date } fecha_hora_salida | Fecha y hora de salida. Ejemplo: "15/03/2024 12:00"
 * @attribute { string } puerta_embarque Puerta de embarque/carril/anden {string}
 * @attribute { string } token_facturacion | Token de facturacion. Ejemplo: "1ab2c"
 * @attribute { string } categoria | Categoría del boleto. [Completo, Inapam, Medio, Profesor]
 * @attribute { integer } numero_operacion | Número de operación. Ejemplo: 123emplo: 123
 * @attribute { float } total_pago | Total de pago (a dos decimales). Ejemplo: 123.45
 * @attribute { string } metodo_pago | Método de pago. [Efectivo, Tarjeta]
 * @attribute { string } estado | Estado del boleto. [Cancelado, Abordado, Vigente, Caducado]
 */
class Boleto {
    /**
     *
     * @param { any } boleto
     * @returns { Boleto }
     * @description Constructor de la clase Boleto
     * @constructs Boleto
     */
    constructor (boleto) {
        this.viaje_id = boleto.viaje_id;
        this.boleto_id = boleto.boleto_id;
        this.correo_pasajero = boleto.correo_pasajero;
        this.estado = boleto.estado;
        this.nombre_pasajero = boleto.nombre_pasajero;
        this.numero_asiento = boleto.numero_asiento;
        this.categoria = boleto.categoria;
        this.origen = boleto.origen;
        this.destino = boleto.destino;
        this.numero_servicio = boleto.numero_servicio;
        this.fecha_hora_salida = boleto.fecha_hora_salida;
        this.puerta_embarque = boleto.puerta_embarque;
        this.token_facturacion = boleto.token_facturacion;
        this.numero_operacion = boleto.numero_operacion;
        this.total_pago = boleto.total_pago;
        this.metodo_pago = boleto.metodo_de_pago;
    }
}

module.exports = Boleto;

File: server/services/boleto-apple-wallet.api/Modelos/Boleto.cs
using System.Text.Json;

namespace boleto_apple_wallet.api.Modelos;
public partial class Boleto
{
	public string? ViajeId { get; set; }
	public string? BoletoId { get; set; }
	public short Estado { get; set; }
	public string? NombrePasajero { get; set; }
	public int Asiento { get; set; }
	public string? Categoria { get; set; }
	public string? Origen { get; set; }
	public string? Destino { get; set; }
	public string? NumeroServicio { get; set; }
	public string? FechaHoraSalida { get; set; } // Formato: "dd-MM-YYYY HH:mm"
	public string? PuertaEmbarque { get; set; }
	public string? TokenFacturacion { get; set; }
	public string? NumeroOperacion { get; set; }
	public float TotalPago { get; set; }
	public string? MetodoPago { get; set; }

	public Boleto(string? viajeId, string? boletoId, short estado, string? nombrePasajero, int asiento, string? categoria, string? origen, string? destino, string? numeroServicio, string? fechaHoraSalida, string? puertaEmbarque, string? tokenFacturacion, string? numeroOperacion, float totalPago, string? metodoPago)
	{
		ViajeId = viajeId;
		BoletoId = boletoId;
		Estado = estado;
		NombrePasajero = nombrePasajero;
		Asiento = asiento;
		Categoria = categoria;
		Origen = origen;
		Destino = destino;
		NumeroServicio = numeroServicio;
		FechaHoraSalida = fechaHoraSalida;
		PuertaEmbarque = puertaEmbarque;
		TokenFacturacion = tokenFacturacion;
		NumeroOperacion = numeroOperacion;
		TotalPago = totalPago;
		MetodoPago = metodoPago;
	}
	// Sobrecarga de constructor para deserializar JSON
	public Boleto(string? json)
	{
		if (json != null)
		{
			var boleto = JsonSerializer.Deserialize<Boleto>(json);
			ViajeId = boleto.ViajeId ?? "-1";
			BoletoId = boleto.BoletoId ?? "-1";
			Estado = boleto.Estado;
			NombrePasajero = boleto.NombrePasajero ?? "Nombre Pasajero";
			Asiento = boleto.Asiento;
			Categoria = boleto.Categoria ?? "Categoria";
			Origen = boleto.Origen ?? "Origen";
			Destino = boleto.Destino ?? "Destino";
			NumeroServicio = boleto.NumeroServicio ?? "Numero Servicio";
			FechaHoraSalida = boleto.FechaHoraSalida ?? "dd-MM-YYYY HH:mm";
			PuertaEmbarque = boleto.PuertaEmbarque ?? "Puerta Embarque";
			TokenFacturacion = boleto.TokenFacturacion ?? "Token Facturacion";
			NumeroOperacion = boleto.NumeroOperacion ?? "Numero Operacion";
			TotalPago = boleto.TotalPago;
			MetodoPago = boleto.MetodoPago ?? "Metodo Pago";
		}
	}
}
